### ISSUE TITLE
Select-Object -ExpandProperty and -Property should be mutually exclusive

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#6161]
[Closed, because it is based on a misconception of how `-ExpandProperty` works.]

`-ExpandProperty` makes `Select-Object` return a single property _value_, which is mutually exclusive with `-Property`, which outputs a _custom object_ with the specified properties.

Currently, `-Property` is quietly ignored when `-ExpandProperty` is also specified.

Steps to reproduce
------------------

```powershell
[pscustomobject] @{ one = 1; two = 2; three = 3 } | 
  Select-Object -ExpandProperty one -Property three
```

Expected behavior
-----------------

A statement-terminating error complaining about mutually exclusive parameters should be issued.

Actual behavior
---------------

```none
1
```

That is, the `-Property` argument was silently _ignored_.

Environment data
----------------

```powershell
PowerShell Core v6.0.1
Windows PowerShell v5.1
```

**Comment by dragonwolf83** [comment]
That is not entirely correct and making it mutually exclusive would be a breaking change.

`-ExpandProperty` on a Property containing an _Array_  or single value will ignore other properties like your example.

`-ExpandProperty` on a Property containing a _PSObject_ it will add a row for each `-Property` and include the expanded properties that matched to that row.

Using your example, I embed `PSObject` called `three` containing values 3 and 4 and it gets added :
```PowerShell
[pscustomobject] @{ one = 1; two = 2; three = [pscustomobject] @{ three = 3; four = 4 } }| 
    Select one, two -ExpandProperty three 

three four one two
----- ---- --- ---
    3    4   1   2
```

This is very useful for flattening data rom nested objects so that you have it all in the main record. For example, I took custom fields that were embedded in a ticket and expanded it this way so that the custom field would look like a regular field alongside the others.

This is one of the things I do a lot so the proposed change would break half my scripts.

**Comment by mklement0** [comment]
Thanks, @dragonwolf83 - I forgot about the many subtleties of `-ExpandProperty`.

The current `Select-Object` docs are woefully inadequate (incomplete, and partially misleading).

Here's what actually happens, and from that perspective the behavior is consistent in all 3 cases you mention:

* The single property you pass to `-ExpandProperty` becomes the _output type_. That is, instances of whatever type the referenced property is are output.

* Any properties you pass to `-Property` are _added as  note properties_ to those output-type instances.   
Often, the output formatting may not show these note properties, which makes it _look_ like they're not there (ignored), but they are there.

In short: The real problem is one of _poor documentation_, and I'll add an issue in the docs repo.

**Comment by mklement0** [comment]
I've created a documentation issue: https://github.com/PowerShell/PowerShell-Docs/issues/2130
